pkg/utils/math.go

```diff
@@ -0,0 +1,20 @@
+package utils
+
+import "sort"
+
+// MedianFloat32 gets median value for an array of float32
+func MedianFloat32(input []float32) float32 {
+	num := len(input)
+	if num == 0 {
+		return 0
+	}
+	sort.Slice(input, func(i, j int) bool {
+		return input[i] < input[j]
+	})
+	if num%2 != 0 {
+		return input[num/2]
+	}
+	left := input[(num-1)/2]
```

Comment:
I am pretty sure this should be:

```go
left := input[num/2 - 1]
right := input[num/2]
```

And if we do that we need another early return:

```go
if num == 0 {
  return 0
} else if num == 1 {
  return input[num]
}
```

The current behavior [fails on odd numbers](https://go.dev/play/p/DtscbuDRSCK)

Reply:
The odd number case is covered above. So both works (what Shishir had and what you suggested) for even numbers.